### ISSUE TITLE
Incorrect content-length calculation (length is always -1)

### ISSUE THREAD

**Issue by frmatthew** [flutterchina/dio#1060]
### New Issue Checklist

- [X] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | flutter                          |      |
| Dio Version                    | 4.0.0-beta3                                  |      |

### Issue Description and Steps

[transformer.dart](https://github.com/flutterchina/dio/blob/master/dio/lib/src/transformer.dart) has a bug in its length calculation. See the following lines, which always set the length at -1 and discard the result of int.parse:

```
length = -1;
int.parse(response.headers[Headers.contentLengthHeader]?.first ?? '-1');
```

The end result is that the length/total is always -1, making download progress impossible to track accurately. I would submit a PR (`length = int.parse(response.headers[Headers.contentLengthHeader]?.first ?? '-1');` I presume), but I can't tell if this was accidental or a temporary hack. Looks like the issue was introduced [here](https://github.com/flutterchina/dio/commit/10d071ec813ae27cfb2c43d3bfc50e1927501035#diff-1bc936903366c430e1c09b039066a0a2cc61123216fa6525316bdaeb1ebfb470).

Thanks.

**Comment by wendux** [comment]
It is fixed in  v4.0.0-beta4

**Comment by panchaoco** [comment]
I also encountered such a problem
